tests/system-tests/rdscore/internal/rdscorecommon/sriov-rootless-dpdk.go

```diff
@@ -0,0 +1,861 @@
+		serverDeploymentLabelMap)
+
+	if err != nil {
+		glog.V(100).Infof("Failed to create server deployment %s in namespace %s: %v",
+			serverDPDKDeploymentName, deploymentNamespace, err)
+
+		return fmt.Errorf("failed to create server deployment %s in namespace %s: %w",
+			serverDPDKDeploymentName, deploymentNamespace, err)
+	}
+
+	return nil
+}
+
+// CleanupRootlessDPDKServerDeployment cleaning up the rootless dpdk server deployment.
+func CleanupRootlessDPDKServerDeployment() {
+	By("Ensuring rootless DPDK server deployment was deleted")
+
+	err := cleanUpRootlessDPDKDeployment(APIClient, serverDPDKDeploymentName, deploymentNamespace, serverPodLabel)
+	Expect(err).ToNot(HaveOccurred(),
+		fmt.Sprintf("Failed to cleanup deployment %s from the namespace %s: %v",
+			serverDPDKDeploymentName, deploymentNamespace, err))
+}
+
+func cleanUpRootlessDPDKDeployment(
+	apiClient *clients.Settings,
+	deploymentName,
+	nsName,
+	podLabel string) error {
+	if deploymentName == "" {
+		glog.V(100).Infof("The rootless dpdk deployment name has to be provided")
+
+		return fmt.Errorf("the rootless dpdk deployment name has to be provided")
+	}
+
+	if nsName == "" {
+		glog.V(100).Infof("The rootless dpdk deployment namespace has to be provided")
+
+		return fmt.Errorf("the rootless dpdk deployment namespace has to be provided")
+	}
+
+	_, err := deployment.Pull(apiClient, deploymentName, nsName)
+
+	if err != nil {
+		glog.V(100).Infof("Deployment %s not found in namespace %s, %v", deploymentName, nsName, err)
+	}
+
+	glog.V(100).Infof("Ensure %s deployment does not exist in namespace %s", deploymentName, nsName)
+
+	err = apiobjectshelper.DeleteDeployment(apiClient, deploymentName, nsName)
+
+	if err != nil {
+		glog.V(100).Infof("Failed to delete deployment %s from nsname %s due to %v",
+			deploymentName, nsName, err)
+
+		return fmt.Errorf("failed to delete deployment %s from nsname %s due to %w",
+			deploymentName, nsName, err)
+	}
+
+	err = apiobjectshelper.EnsureAllPodsRemoved(apiClient, nsName, podLabel)
+
+	if err != nil {
+		glog.V(100).Infof("Failed to delete pods in namespace %s with the label %s: %v", nsName, podLabel, err)
+
+		return fmt.Errorf("failed to delete pods in namespace %s with the label %s: %w", nsName, podLabel, err)
+	}
+
+	return nil
+}
+
+func defineAndCreateDPDKDeployment(
+	apiClient *clients.Settings,
+	deploymentName,
+	deploymentNamespace,
+	scheduleOnHost,
+	saName string,
+	securityContext corev1.SecurityContext,
+	podSC *corev1.PodSecurityContext,
+	podNetConfig []*types.NetworkSelectionElement,
+	containerCmd []string,
+	deployLabels map[string]string) error {
+	glog.V(100).Infof("Creating container %s definition", deploymentName)
+
+	nodeSelector := map[string]string{"kubernetes.io/hostname": scheduleOnHost}
+
+	dpdkContainerCfg, err := defineDPDKContainer(deploymentName, containerCmd, securityContext)
+
+	if err != nil {
+		glog.V(100).Infof("Failed to set dpdk container %s due to %v", deploymentName, err)
```

Comment:
nit: DPDK is abbreviation so should use capital letters 
```suggestion
		glog.V(100).Infof("Failed to set DPDK container %s due to %v", deploymentName, err)
```